### PR TITLE
Deserialize manifest protos

### PR DESCRIPTION
**Pull request by markww**

### CODE REVIEW

File: src/transform/ir_proto_to_datalog/arcs_manifest_tree/recipe.cc [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+#include <memory>
+#include <string>
+#include <cstdint>
+
+namespace raksha::transform::arcs_manifest_tree {
+
+namespace logging = ::raksha::transform::logging;
+
+Recipe::Recipe(const absl::string_view name, const uint64_t num_handles)
```
Comment (bgogul) [inline src/transform/ir_proto_to_datalog/arcs_manifest_tree/recipe.cc:16]: I don't think we should be replicating the arcs manifest as is in C++. We want to introduce a level of abstraction like graph (which is that the recipe is).

File: src/transform/ir_proto_to_datalog/logging/context_logger.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
+#ifndef SRC_TRANSFORM_IR_PROTO_TO_DATALOG_LOGGING_CONTEXT_LOGGER_H_
+#define SRC_TRANSFORM_IR_PROTO_TO_DATALOG_LOGGING_CONTEXT_LOGGER_H_
+
+#include "absl/strings/string_view.h"
```
Comment (bgogul) [inline src/transform/ir_proto_to_datalog/logging/context_logger.h:4]: Can't we use https://github.com/google/glog?

**Review by bgogul — COMMENTED** [review]
As I mentioned in the comments, we don't want a 1-1 mapping from everything in `manifest.proto` to C++ data structure. Let us chat a bit more offline.

File: src/transform/ir_proto_to_datalog/logging/context_logger.h (hunk 2 above)
Comment (markww) [reply src/transform/ir_proto_to_datalog/logging/context_logger.h:4]: Thank you for this pointer, I will use that library.

**Comment by markww** [comment]
This PR was never destined for submission directly; it was always a bit of a scrapheap to dig through for usable bits of code for more polished and focused PRs. I think, at this point, it's entirely picked over and it's time to close it.

**Pull request closed without merging**